Add lowpass branch rebuild 2

Pall-v opened the pull request:
rebuild changes from previous pull request

Pall-v committed:
drafted highpass filter fn.

Pall-v committed:
setup.py and test_high_pass_filter.py will now run

Pall-v committed:
Corrections to high-pass finter function and added assertion error message to its test script

Pall-v committed:
line not needed for pytest.  e.g. with: pytest test_high_pass_filter.py

Pall-v committed:
Changes related to using a low pass filter since that was simpler to implement.

Pall-v committed:
changes related to switchong to a low-pass filter

sedv8808 committed:
Merge pull request #16 from UBC-MDS/add_high_pass_fn

Add high pass fn

sedv8808 committed:
Revert "Add high pass fn"

Pall-v committed:
Merge pull request #17 from UBC-MDS/revert-16-add_high_pass_fn

Revert "Add high pass fn"

Pall-v committed:
rebuilding changes from previous pull request

Pall-v committed:
Revert "rebuilding changes from previous pull request"

This reverts commit c37e4711db1dd1e2a4a88b6c5717551bfd2f1273.

Pall-v committed:
Revert "Revert "rebuilding changes from previous pull request""

This reverts commit e37c7f030200108213d6f5b13d1154add4add228
to exclude pytest cache files

mchiriboga commented on tests/test_low_pass_filter.py:
```diff
@@ -0,0 +1,13 @@
+
+input_signal = np.genfromtxt('data/bark.csv', dtype = 'float32')
+
+def test_low_pass_filter_attenuates_frequencies_below_1000_cutoff():
```
I think the name of the test should now say `above_1000_cutoff` instead of `below_1000_cutoff`, since the filter is now low pass.

mchiriboga commented on tests/test_low_pass_filter.py:
```diff
@@ -0,0 +1,13 @@
+input_signal = np.genfromtxt('data/bark.csv', dtype = 'float32')
+
+def test_low_pass_filter_attenuates_frequencies_below_1000_cutoff():
+    expected_output = np.genfromtxt('data/lowpass/bark_highpass_1000Hz_24db.csv', dtype = 'float32')
```
I think the name of the file should be `bark_lowpass_1000Hz_24db.csv` instead of `bark_highpass_1000Hz_24db.csv`

Pall-v replied:
Yes, I left the filename the same because it needs to be replaced with a completely different file.  I opened an issue for that.

Pall-v replied:
Filename is now changed.

sedv8808 replied:
Hi Paul, just change description at the Readme file.

Pall-v committed:
Replaced highpass csv test file with lowpass test file

Also removed highpass wav because my new process for generating the test file didn't involve a .wav

Pall-v committed:
point to test to lowpass test data file

Pall-v committed:
Lowpass test changes

Now uses mean squared error to compare instead of requiring exact match.

Pall-v committed:
lowpass filter test data corrections

- Corrected lowpass test data
- Removed db spec from filemane since that is not relevant to the current inmplementation

Pall-v committed:
Update test_low_pass_filter.py

corr

Pall-v committed:
Add exception for cutoff frequency

pytest.raises(Exception) does not seem to be working

Pall-v commented:
Made a correction to the test data and changed the comparison method because the test was failing.

Pall-v committed:
Update README.md

Changed from high to low pass filter

Pall-v committed:
Merge branch 'master' into add_lowpass_rebuild_2
